vector/src/main/java/im/vector/app/features/navigation/DefaultNavigator.kt

```diff
@@ -355,54 +329,22 @@ class DefaultNavigator @Inject constructor(
     }
 
     override fun openCreateDirectRoom(context: Context) {
-        val intent = when (val currentGroupingMethod = appStateHandler.getCurrentRoomGroupingMethod()) {
-            is RoomGroupingMethod.ByLegacyGroup -> {
-                CreateDirectRoomActivity.getIntent(context)
-            }
-            is RoomGroupingMethod.BySpace -> {
-                if (currentGroupingMethod.spaceSummary != null) {
-                    SpacePeopleActivity.newIntent(context, currentGroupingMethod.spaceSummary.roomId)
-                } else {
-                    CreateDirectRoomActivity.getIntent(context)
-                }
-            }
-            else -> null
-        } ?: return
-        context.startActivity(intent)
+        when (val currentSpace = appStateHandler.getCurrentSpace()) {
+            null -> CreateDirectRoomActivity.getIntent(context)
+            else -> SpacePeopleActivity.newIntent(context, currentSpace.roomId)
```

Comment:
I'm a bit confused here as the function says `openCreateDirectRoom`, but within a space it looks like it's doing something different (same for the function higher above). I know this was also the way it was in the old code but maybe now is a good time to improve upon it

Reply:
In space it opens a list of users in this space while at home it opens an activity to search for public users. So it's quite similar and it's part of "create dm" flow. I agree that naming is confusing a bit, but not sure if we can do something better here

Reply:
I see, it makes sense. Maybe a comment here mentioning that `SpacePeopleActivity` is where direct rooms are created within a space? WDYT?

Reply:
AFAIU `SpacePeopleActivity` and `SpacePeopleFragment` are not only used to create DMs. But `People` could be renamed to `User` probably.